geekbench-calculate-scores.py

```diff
@@ -0,0 +1,221 @@
+
+  wl_units = {"AES" : rate_units['BytesSecond'],
+  "Twofish" : rate_units['BytesSecond'], "SHA1" : rate_units['BytesSecond'],
+  "SHA2" : rate_units['BytesSecond'],
+  "BZip2 Compress" : rate_units['BytesSecond'],
+  "BZip2 Decompress" : rate_units['BytesSecond'],
+  "JPEG Compress" :  rate_units['PixelsSecond'],
+  "JPEG Decompress" : rate_units['PixelsSecond'],
+  "PNG Compress" : rate_units['PixelsSecond'],
+  "PNG Decompress" : rate_units['PixelsSecond'],
+  "Sobel" : rate_units['PixelsSecond'],
+  "Lua" : rate_units['BytesSecond'] ,
+  "Dijkstra" : rate_units['PairsSecond'],
+  "BlackScholes" : rate_units['NodesSecond'],
+  "Mandelbrot" : rate_units['Flops'],
+  "Sharpen Filter" :   rate_units['Flops'],
+  "Blur Filter" :  rate_units['Flops'],
+  "SGEMM" : rate_units['Flops'], "DGEMM" :  rate_units['Flops'],
+  "SFFT" : rate_units['Flops'], "DFFT" :  rate_units['Flops'],
+  "N-Body" : rate_units['PairsSecond'],
+  "Ray Trace" : rate_units['PixelsSecond'],
+  "Stream Copy" : rate_units['BytesSecond'],
+  "Stream Scale" : rate_units['BytesSecond'],
+  "Stream Add" :  rate_units['BytesSecond'],
+  "Stream Triad" : rate_units['BytesSecond'] }
+
+  prefixes = ["", "K", "M", "G", "T"]
+
+  wl_unit = wl_units[wl_name]
+  divisor = 1000.0
+
+  if (wl_unit == "B/sec"):
+    divisor = 1024.0
+
+  for prefix in range(0,4):
+    if rate < divisor:
+      break
+    rate /= divisor  
+
+  if rate > 10.0:
+    rate = round(rate, 1)
+  else:
+    rate = round(rate, 2)
+
+  return "%s %s%s" % (rate, prefixes[prefix], wl_unit)
+
+def main():
+  jsonfile = open(sys.argv[1])
+  jsonobj = json.load(jsonfile)
+
+  sections = get_sections(jsonobj)
+  gb_sc_score = compute_geekbench_sc_score(sections)
+  gb_mc_score = compute_geekbench_mc_score(sections)
+
+
+  for (section, wls) in sections.iteritems():
+    print section
+    for wl in wls:
+      print "\t%s".expandtabs(2) % wl.name
+
+      wl_sc_score_str = str(int(wl.wl_sc_score())).rjust(12)
+      sc_rate_str = get_rate_string(wl.name, wl.sc_rate()).rjust(21)
+      print "\tsingle-core%s%s".expandtabs(4) % (wl_sc_score_str, sc_rate_str)
+
+      wl_mc_score_str = str(int(wl.wl_mc_score())).rjust(13)
+      mc_rate_str = get_rate_string(wl.name, wl.mc_rate()).rjust(21)
+      print "\tmulti-core%s%s".expandtabs(4) % (wl_mc_score_str , mc_rate_str)
+    print ""
+
+  print "Benchmark Summary"
+  for section in sections.iterkeys():
+    section_score_str = str(section + " Score").ljust(24)
+    sect_sc_score_str = str(compute_sect_sc_score(sections[section])).rjust(7)
+    sect_mc_score_str = str(compute_sect_mc_score(sections[section])).rjust(7)
+    print "\t%s%s%s".expandtabs(2) % (section_score_str, sect_sc_score_str, 
+    sect_mc_score_str)
+  print ""
+
+  gb_sc_score_just = str(gb_sc_score).rjust(7)
+  gb_mc_score_just = str(gb_mc_score).rjust(7)
+
+  print "\t%s%s%s".expandtabs(2) % ("Geekbench Score".ljust(24),
+  gb_sc_score_just, gb_mc_score_just)
+  jsonfile.close()
```

Comment:
It's a good idea to close the file as soon as you're done reading it. You could do it immediately after the `jsonobj = json.load(jsonfile)` line.